### ISSUE TITLE
When multiple devices are running in parallel, some devices may report errors, some devices may execute normally, and the error reporting devices may also execute normally independently

### ISSUE THREAD

**Issue by suzhenyu006** [appium/appium#18405]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When multiple devices are running in parallel, some devices may report errors, some devices may execute normally, and the error reporting devices may also execute normally independently

### Expected Behavior

Parallel execution of multiple devices is normal

### Minimal Reproducible Example

1. Create 4-5 desired_capabilities for devices with different udids, and other settings can be the same as below:

{'platformName': 'Android', 'deviceName': 'DUK-AL20', 'platformVersion': '9', 'newCommandTimeout': '3600', 'automationName': 'UIAutomator2', 'uiautomator2ServerInstallTimeout': 120000, 'adbExecTimeout': 120000, 'app': '', 'appPackage': None, 'appActivity': None, 'browserName': 'Chrome', 'udid': '6EB0217805005163', 'unlockType': 'pin', 'noReset': True, 'showChromedriverLog': True, 'fullReset': Fal se, 'skipServerInstallation': True, 'unlockKey': '123456', 'appium:chromedriverChromeMappingFile': '/Users/sohu/Desktop/device_ center/sonic/agent/appiumChromeDrivers/version.json', 'appium:chromedriverExecutableDir': '/Users/sohu/Desktop/device_ center/sonic/agent/appiumChromeDrivers'}

2. Then start the following python script:

port = get_ port()

self.driver = appiumdriver.Remote(command_executor=" http://10.3.21.123: "+ self.port, desired_capabilities =self.desired_caps)

self.driver.get(' https://www.google.com ')

stop_ appium_ server()

The following is my java-client code for starting the appium server, which is very simple:

    public static int start(String udId) {
        close(udId);
        AppiumDriverLocalService service = AppiumDriverLocalService.buildService(new AppiumServiceBuilder()
                .withIPAddress("0.0.0.0")
                .usingAnyFreePort()
                .withArgument(GeneralServerFlag.LOG_LEVEL, "debug")
                .withArgument(GeneralServerFlag.ALLOW_INSECURE, "chromedriver_autodownload"));
        System.out.println("start!!!");
        service.start();
        logger.info("Appium server start...");
        serviceMap.put(udId,service);
        return service.getUrl().getPort();
    }

    public static void close(String udId) {
        AppiumDriverLocalService service = serviceMap.get(udId);
        if (service != null && service.isRunning()) {
            service.stop();
            logger.info("Appium server stop...");
        }
        serviceMap.remove(udId);
        try
        {
            Runtime.getRuntime().exec("adb -s " + udId + " shell am force-stop io.appium.uiautomator2.server");
            Runtime.getRuntime().exec("adb -s " + udId + " shell am force-stop io.appium.uiautomator2.server.test");
        }
        catch (Exception e)
        {
            System.out.println(e.getMessage());
        }
        System.out.println("close!!!");
    }

### Environment

- Operating system:
- If running via `appium` CLI...
  - Appium CLI version (output of `appium --version`): 2.0.0-beta.57
  - Node.js version (output of `node --version`): v16.15.0
  - `npm` version (output of `npm --version`): 8.5.5
  - Last `appium` version which did _not_ exhibit the problem: 
- If running Appium Desktop, its version:
- Appium driver(s) and their version(s): appium-java-client: 8.3.0 / appium-python-client: 2.4.0
- Appium plugin(s) and their version(s): uiautomator2@2.12.6 [installed (NPM)] / xcuitest@4.19.1 [installed (NPM)]
- Platform and version under test: /
- Real device or emulator/simulator: All real devices

### Link to Appium Logs

https://gist.github.com/suzhenyu006/496c2db31be1d77bc30a9d9c88695e7a

### Futher Information

Due to the fact that there are multiple devices and the log is all inclusive, the following key device IDs are highlighted:

5LM0216111007911

6EB0217805005163

ff16bbfc

**Comment by mykola-mokhnach** [comment]
Do you follow https://github.com/appium/appium-uiautomator2-driver#parallel-tests to setup session capabilities. Also, make sure you use the most recent uia2 version. There was an issue with automated chromedrivers download

**Comment by suzhenyu006** [comment]
Thanks Reply

I am using uiaDriver and uia-server are relatively new

I deleted the 'chromedriverChromeMappingFile' in the configuration, added the 'chromedriverPorts': [10700, [10710, 10810]] configuration, and then my chormedriver problem disappeared, and I got a new error:

```
Message: An unknown server-side error occurred while processing the command. Original error: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: socket hang up

Stacktrace:

UnknownError: An unknown server-side error occurred while processing the command. Original error: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: socket hang up

at getResponseForW3CError (/usr/local/lib/node_modules/appium/node_modules/@appium/base-driver/lib/protocol/errors.js:1067:9)

at asyncHandler (/usr/local/lib/node_modules/appium/node_modules/@appium/base-driver/lib/protocol/protocol.js:476:57)
```

Suspected to be io.appium.uiatuomator2.server problem in the phone, but I don't know how to solve it?

uia2 driver version that i use is 


So please see what I said above. Every time I start APPIUM, I will specify a unique port number.

```
self.driver = appiumdriver.Remote(command_executor=" http://10.3.21.123/: "+ self.port, desired_capabilities =self.desired_caps)
```

 Do I still need to give SystemPort cap? Is it consistent if required? Is it a problem that I didn't specify the SystemPort?

**Comment by mykola-mokhnach** [comment]
Please create a new issue report for your problem. Closing this one as resolved
